### ISSUE TITLE
In manual testing on CIFAR-10 example, prediction error of test set was 90.88%. What happened?

### ISSUE THREAD

**Issue by everpeace** [BVLC/caffe#1405]
Hi folks.

I followed [this CIFAR-10 tutorial](http://caffe.berkeleyvision.org/gathered/examples/cifar10.html) and got trained model `cifar10_quick_iter_5000.caffemodel`.

In this training, caffe output accuracy on test data set was 

```
(git:master) $ tail train_log.txt
I1105 00:57:46.100651 2106102528 solver.cpp:206]     Train net output #0: loss = 0.482015 (* 1 = 0.482015 loss)
I1105 00:57:46.100685 2106102528 solver.cpp:403] Iteration 4900, lr = 0.0001
I1105 00:58:09.639089 2106102528 solver.cpp:317] Snapshotting to examples/cifar10/cifar10_quick_iter_5000.caffemodel
I1105 00:58:09.645743 2106102528 solver.cpp:324] Snapshotting solver state to examples/cifar10/cifar10_quick_iter_5000.solverstate
I1105 00:58:09.727504 2106102528 solver.cpp:228] Iteration 5000, loss = 0.565611
I1105 00:58:09.727572 2106102528 solver.cpp:247] Iteration 5000, Testing net (#0)
I1105 00:58:16.502969 2106102528 solver.cpp:298]     Test net output #0: accuracy = 0.7491
I1105 00:58:16.503042 2106102528 solver.cpp:298]     Test net output #1: loss = 0.74817 (* 1 = 0.74817 loss)
I1105 00:58:16.503067 2106102528 solver.cpp:233] Optimization Done.
I1105 00:58:16.503078 2106102528 caffe.cpp:121] Optimization Done.
```

However, once I tried to predict test data set on ipython,  error rate was extremely high (that was 90.88 percent).  My python notebook which I used on this experiment is [here](http://nbviewer.ipython.org/github/everpeace/caffe/blob/master/examples/cifar10_test_error_is_high.ipynb).

I'm not sure what went wrong.  I'm really appreciated if someone gave me help.
